collect_app/src/main/java/org/odk/collect/android/widgets/utilities/GeoWidgetUtils.java

```diff
@@ -23,16 +23,22 @@ private GeoWidgetUtils() {
 
     }
 
+    //Cm accuracy #4198
     public static String getGeoPointAnswerToDisplay(Context context, String answer) {
         try {
             if (answer != null && !answer.isEmpty()) {
                 String[] parts = answer.split(" ");
-                return context.getString(
-                        R.string.gps_result,
+                //Guard against bad answer
+                if (parts.length < 4) {
```

Comment:
I don't think we expect this at all and probably want a non-silent failure in the case that it did happen. We can take this check out for the moment.